controller/controller-http/src/main/java/com/pinterest/rocksplicator/controller/util/Result.java

```diff
@@ -0,0 +1,64 @@
+/*
+ *  Copyright 2017 Pinterest, Inc.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package com.pinterest.rocksplicator.controller.util;
+
+/**
+ * A data structure for holding the result from TaskQueue so that
```

Comment:
the documentation does not say the String message field is available to clients. you cna check the http://docs.oracle.com/javaee/7/api/javax/ws/rs/WebApplicationException.html#WebApplicationException-javax.ws.rs.core.Response-
public WebApplicationException(Response response), this response that will be returned to the client, which the front end can retrieve.

So basically, for all the resource i found, in order to pass the message to front end, i need to create a Response with the message and then do throw WebApplicationException (Response).  https://stackoverflow.com/questions/7998390/how-do-i-set-a-40x-error-with-a-custom-message-on-jax-rs-exception

Also, somethings the response are not necessarily a WebApplicationException. When thats the case, we still need to put the message in a Response object. 

Let me know if you have a better solution